Library/Homebrew/brew.sh

```diff
@@ -443,23 +443,25 @@ GIT_REVISION=$("${HOMEBREW_GIT}" -C "${HOMEBREW_REPOSITORY}" rev-parse HEAD 2>/d
 # safe fallback in case git rev-parse fails e.g. if this is not considered a safe git directory
 if [[ -z "${GIT_REVISION}" ]]
 then
-  GIT_HEAD="$(cat "${HOMEBREW_REPOSITORY}/.git/HEAD" 2>/dev/null)"
+  read -r GIT_HEAD <"${HOMEBREW_REPOSITORY}/.git/HEAD" 2>/dev/null
   if [[ "${GIT_HEAD}" == "ref: refs/heads/master" ]]
   then
-    GIT_REVISION="$(cat "${HOMEBREW_REPOSITORY}/.git/refs/heads/master" 2>/dev/null)"
+    read -r GIT_REVISION <"${HOMEBREW_REPOSITORY}/.git/refs/heads/master" 2>/dev/null
   elif [[ "${GIT_HEAD}" == "ref: refs/heads/stable" ]]
   then
-    GIT_REVISION="$(cat "${HOMEBREW_REPOSITORY}/.git/refs/heads/stable" 2>/dev/null)"
+    read -r GIT_REVISION <"${HOMEBREW_REPOSITORY}/.git/refs/heads/stable" 2>/dev/null
   fi
   unset GIT_HEAD
 fi
 
 if [[ -n "${GIT_REVISION}" ]]
 then
   GIT_DESCRIBE_CACHE_FILE="${GIT_DESCRIBE_CACHE}/${GIT_REVISION}"
-  if [[ -r "${GIT_DESCRIBE_CACHE_FILE}" ]]
+  # Pass `--` to guard against `git` being confused by a file named `HEAD`.
+  if "${HOMEBREW_GIT}" -C "${HOMEBREW_REPOSITORY}" diff-index --quiet --exit-code HEAD -- 2>/dev/null &&
+     [[ -r "${GIT_DESCRIBE_CACHE_FILE}" ]]
```

Comment:
```suggestion
  if [[ -r "${GIT_DESCRIBE_CACHE_FILE}" ]] && \
     "${HOMEBREW_GIT}" -C "${HOMEBREW_REPOSITORY}" diff --quiet HEAD -- 2>/dev/null &&
```
- much less expensive to check if the file is readable so do that first
- use `git diff` unless it's slower
- `--quiet` implies `--exit-code`

Reply:
If you use `git diff` then remember to pass `--no-ext-diff` too.

Reply:
Done! Switched the order and used `git diff` instead.